component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java

```diff
@@ -489,20 +489,39 @@ private void proceedWithOTP(HttpServletResponse response, AuthenticationContext
      */
     private void checkStatusCode(HttpServletResponse response, AuthenticationContext context,
                                  String queryParams, String errorPage) throws AuthenticationFailedException {
+        String screenValue = null;
         boolean isRetryEnabled = SMSOTPUtils.isRetryEnabled(context, getName());
         String loginPage = getLoginPage(context);
         String url = getURL(loginPage, queryParams);
+        String username = String.valueOf(context.getProperty(SMSOTPConstants.USER_NAME));
+        String tenantDomain = MultitenantUtils.getTenantDomain(username);
+        String tenantAwareUsername = MultitenantUtils.getTenantAwareUsername(username);
+        UserRealm userRealm = SMSOTPUtils.getUserRealm(tenantDomain);
+        try {
+            screenValue = getScreenAttribute(context, userRealm, tenantAwareUsername);
+        } catch (UserStoreException e) {
+            throw new AuthenticationFailedException("Failed to get the user from user store. ", e);
```

Comment:
Did the error message matches the scenario? Here we are getting screen attribute but the error message says "Failed to get the user from userstore. Further shall we please include the context specific information such as tenantAware UserName